### ISSUE TITLE
Peer verification failed

### ISSUE THREAD

**Issue by namtsui** [flyinghead/flycast#388]
**Please Note: This form is the minimum required information for submitting bugs.**  
**Removing this form may lead to your issue being closed until it is completed.**

Platform / OS / Hardware:
windows 6800 xt

Github hash:
99bf7f77c433d85841237c96a7e1434806bf6c27 2021-09-29 works

Hardware: 

**_Description of the Issue_**

Peer verification fails during netplay while testing the net-rollback branch. I saw on discord that peer verification means the following:

```flyinghead: If peer verification is failing, your opponent is probably running a different Flycast dojo build, or using a different rom, or a different savestate```

However, I used the same files for cvs2. An older version works but more recent builds fail with peer verification failing. This leads me to believe that there was a regression.

2021-09-29 netplay works on 1) openbsd to windows opponent and  on 2) windows to windows opponent.

**_Debugging Steps Tested_**

I bisected by having my friend and I run matching available windows builds. Not all of these have builds. I marked BAD as the ones that exhibited the problem and GOOD as the one that worked. The offending commit is somewhere the range of 99bf7f7 (good) and 1072235 (bad).

c08f177 2021-10-14T10:03:48.000Z BAD
1c6e722 2021-10-14T09:51:24.000Z
f9d5b8e 2021-10-12T15:41:20.000Z
163888f 2021-10-10T15:33:44.000Z
c2caef4 2021-10-07T18:10:10.000Z BAD
bc238de 2021-10-06T17:00:01.000Z
1072235 2021-10-04T17:47:35.000Z BAD
8183bf7 2021-10-03T18:51:26.000Z
82bec1e 2021-10-03T18:28:03.000Z
897cc01 2021-10-03T16:43:39.000Z
c1bcf91 2021-10-01T14:54:55.000Z
699dcda 2021-09-30T13:12:15.000Z
99bf7f7 2021-09-29T08:32:17.000Z GOOD

**_Logs Gathered_**

**_Screenshots_**

(Replace this line with any issue screenshots)

**Comment by flyinghead** [comment]
Like I suspected on Discord this was in fact part of a larger problem, one that could lead to crashes. For some reason it never happened during my tests. This should fix it: https://github.com/flyinghead/flycast/commit/7ecb6a6d967ff42df34c43a43b008bf8df1c53ba

Great job bisecting and thanks a lot for your detailed reporting!!

**Comment by namtsui** [comment]
Yeah, once we switched to flycast dojo, we experienced some random crashes mid-match in cvs2. I had forgotten to mention that. Nice intuition and thanks for the fix.

I haven't tested it yet, but I will report back.
